django_cloud_deploy/cli/prompt_manager.py

```diff
@@ -0,0 +1,673 @@
+
+    """
+
+    # Parameter Manager Owns must be set
+    PARAMETER = None
+
+    def __init__(self, console: io.IO, step: str, args):
+        self.console = console
+        self.step = step
+        self.args = args
+
+    def prompt(self) -> None:
+        pass
+
+    def validate(self, val: Any) -> bool:
+        return True
+
+    def get_credentials(self):
+        return self.args.get('credentials', None)
+
+    def _is_valid_passed_arg(self) -> bool:
+        value = self.args.get(self.PARAMETER, None)
+        if value is None:
+            return False
+
+        try:
+            if self.validate(value):
+                msg = '{} {}: {}'.format(self.step, self.PARAMETER, value)
+                self.console.tell(msg)
+                return True
+        except ValueError as e:
+            self.console.error(e)
+            return False
+
+        return False
+
+
+class StringTemplatePromptManager(TemplatePromptManager):
+    """Template for a simple string Prompt Manager."""
+
+    PARAMETER = ''
+    PARAMTER_PRETTY = ''
+    DEFAULT_VALUE = ''
+    BASE_MESSAGE = '{} Enter a value for {} or leave blank to use'
+    DEFAUlT_MESSAGE = '[{}]: '
+
+    def _generate_name_prompt(self):
+        base_message = self.BASE_MESSAGE.format(self.step, self.PARAMTER_PRETTY)
+        default_message = self.DEFAUlT_MESSAGE.format(self.DEFAULT_VALUE)
+        msg = '\n'.join([base_message, default_message])
+        return prompt_new.AskPrompt(
+            msg, self.console, self.validate, default=self.DEFAULT_VALUE)
+
+    def prompt(self):
+        if self._is_valid_passed_arg():
+            return
+
+        self.args[self.PARAMETER] = self._generate_name_prompt().prompt()
+
+
+class GoogleProjectNameManager(TemplatePromptManager):
+
+    PARAMETER = 'project_name'
+
+    def __init__(self,
+                 console: io.IO,
+                 step: str,
+                 args,
+                 project_client: Optional[project.ProjectClient] = None):
+        super().__init__(console, step, args)
+        creds = self.get_credentials()
+        self.project_client = (project_client or
+                               project.ProjectClient.from_credentials(creds))
+
+    def validate(self, s: str) -> bool:
+        """Validates that a string is a valid project name.
+
+        Args:
+            s: The string to validate.
+
+        Raises:
+            ValueError: if the input string is not valid.
+        """
+        if not (4 <= len(s) <= 30):
+            raise ValueError(
+                ('Invalid Google Cloud Platform project name "{}": '
+                 'must be between 4 and 30 characters').format(s))
+
+        if self._is_new_project():
+            return True
+
+        project_id = self.args.get('project_id', None)
+        if project_id is None:
+            raise ValueError('Project Id must be set')
+
+        project_name = self.project_client.get_project(project_id)['name']
```

Comment:
So we force the user to specify the same name? Why would we want that? Maybe it should be an error that the flag is set when using an existing project.

Reply:
I agree its wierd. Ill think we can just ignore that they gave it to us and use the one the api call gives us.